openedx_tagging/core/tagging/models/base.py

```diff
@@ -385,6 +385,11 @@ def tag_object(
         Raised ValueError if the proposed tags are invalid for this taxonomy.
         Preserves existing (valid) tags, adds new (valid) tags, and removes omitted (or invalid) tags.
         """
+        if not isinstance(tags, (list, tuple)):
+            raise ValueError(
+                _(f"Tags must be a list or tuple, not {type(tags).__name__}.")
+            )
```

Comment:
Explicit error if not a list or tuple

Reply:
@rpenido In which cases can the tags be a tuple? In my opinion I would just keep it as a list, so as not to complicate the typing here